### PR TITLE
[ISSUE #5779] Use the correct serviceManager version to get the service list

### PR DESCRIPTION
**Pull request by CoderPoet**
## What is the purpose of the change

Fix [ISSUE #5779](https://github.com/alibaba/nacos/issues/5779)

## Brief changelog

- Use the serviceManager v2 to get the service list

## Verifying this change

MCP-Over-XDS successfully synced Nacos services to Istio Envoy XDS

image

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosToMcpResources.java [hunk 1]
```diff
@@ -48,71 +46,78 @@
  */
 @org.springframework.stereotype.Service
 public class NacosToMcpResources {
-    
+
```
Comment (KomachiSion) [inline istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosToMcpResources.java:49]: Please do not change the indent

Please use nacos code style to reformat codes.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosToMcpResources.java (hunk 1 above)
Comment (CoderPoet) [reply istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosToMcpResources.java]: Thanks for reminding, it's done

File: istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosToMcpResources.java [hunk 2]
```diff
@@ -132,36 +143,33 @@ public void run() {
     }
     
     private String convertName(Service service) {
-        if (!Constants.DEFAULT_GROUP.equals(NamingUtils.getGroupName(service.getName()))) {
-            return NamingUtils.getServiceName(service.getName()) + "." + NamingUtils.getGroupName(service.getName())
-                    + "." + service.getNamespaceId();
+        if (!Constants.DEFAULT_GROUP.equals(service.getGroup())) {
+            return service.getName() + "." + service.getGroup();
```
Comment (lzp0412) [inline istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosToMcpResources.java:147]: return service.getName() + "." + service.getGroup()+ "." + service.getNamespace()

Comment (CoderPoet) [reply istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosToMcpResources.java]: Thanks for the reminder, it has been corrected

**Review by lzp0412 — APPROVED** [review]
LGTM

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**